mixins/form.jsx

```diff
@@ -81,14 +86,10 @@ module.exports = {
       currency: currency.code,
       presets: presets.join(",")
     });
+    var newProps = this.state.props;
+    newProps.amount.values.amount = "";
     this.setState({
-      props: {
```

Comment:
Make sure we don't blow away other props, like frequency, when the currency changes. now set state on *just* amount to clear amount when the currency changes.